test/test_energy_storage_device.py

```diff
@@ -322,6 +325,107 @@ def test_accuracy_pycap_simulation(self):
                                         output['voltage'][i-1])
             # update index
             i = i + 1
+    
+    def test_compatibilty_other_dualfoil(self):
+        # PURPOSE: test to see if Dualfoil object works with the
+        #   version of dualfoil without restard (5.2)
+        df = Dualfoil(path=path2, input_name='li-ion.in',
+                      restart_capable=False)
+
+        # Go through evolve tests. Affirm the following:
+        #   1. Variable held constant stays constant
+        #   2. Dependent variable changes in an expected way
+
+        v = df.get_voltage()
+        # make sure voltage works by ensuring it does not 
+        # equal one of the indicators of failure
+        self.assertTrue(v != 0)
+        self.assertTrue(v != -1)
+
+        # 1. constant current (charge)
+        dt = 15.0  # 15 sec
+        c = 5.0   # 5 amperes
+        df.evolve_one_time_step_constant_current(dt, c)
+        v_fin = df.get_voltage()
+        c_fin = df.get_current()
+        # affirm success
+        # current is constant
+        self.assertEqual(c, c_fin)
+        # charging means potential is increasing
+        self.assertTrue(v_fin > v)
+
+        # 2. constant voltage
+        df.reset()
+        v = df.get_voltage()  # volts
+        # (no way to get an accurate initial current)
+        df.evolve_one_time_step_constant_voltage(dt, v)
+        v_fin = df.get_voltage()
+        c_fin = df.get_current()
+        # affirm success
+        # current should be negative and not zero
+        self.assertTrue(c_fin < 0)
+        # voltage is constant
+        self.assertEqual(v, v_fin)
+
+        # 3. constant load
+        df.reset()
+        load = 0.7  # ohms-m2
+        df.evolve_one_time_step_constant_load(dt, load)
+        v_fin = df.get_voltage()
+        c_fin = df.get_current()
+        # load should be positive
+        load_fin = abs(v_fin / c_fin)
+        # load should be close to equal, but not quite...
+        # REASON: roundoff error of current and voltage
+        # SOLUTION: create room for error with `error`
+        error = 2e-2
+        # affirm success
+        self.assertAlmostEqual(load, load_fin, delta=error)
+        
+        # 4. constant power
+        df.reset()
+        p = 15.0  # ohms-m2
+        df.evolve_one_time_step_constant_power(dt, p)
+        v_fin = df.get_voltage()
+        c_fin = df.get_current()
+        # power is always positive
+        p_fin = abs(v_fin * c_fin)
+        # affirm success
+        self.assertAlmostEqual(p, p_fin, delta=error)
+
+    def test_impedance(self):
+        # test the impedance mode of dualfoil5.2
```

Comment:
Same problem. You check that the code runs not that you get the correct results.